plugin/lib/class-php-compatibility-checker.php

```diff
@@ -109,66 +108,105 @@ public function init() {
+				return ! $this->exclude_plugin( $plugin_data );
 			}
+		);
 
-			// Exclude active plugins if only_active = "yes".
-			if ( 'yes' === $this->only_active ) {
-				// Get array of active plugins.
-				$active_plugins = get_option( 'active_plugins' );
+		$active_plugins = get_option( 'active_plugins' );
+
+		// Add "active" attribute.
+		$plugins = array_map(
+			function( $plugin_file, $plugin_data ) use ( $active_plugins ) {
+				$plugin_data['plugin_file'] = $plugin_file;
+				$plugin_data['active']      = in_array( $plugin_file, $active_plugins, true ) ? 'yes' : 'no';
+				return $plugin_data;
+			},
+			array_keys( $plugins ),
+			$plugins
+		);
 
-				if ( ! in_array( $k, $active_plugins, true ) ) {
-					continue;
-				}
-			}
+		$plugin_info = get_site_transient( 'update_plugins' );
 
-			$plugin_file = plugin_dir_path( $k );
+		// Extract real plugin slugs from the update_plugins transient.
+		foreach ( $plugins as $key => $plugin_data ) {
+			$plugin_file = $plugin_data['plugin_file'];
 
-			// Plugin in root directory (like Hello Dolly).
-			if ( './' === $plugin_file ) {
-				$plugin_path = $plugin_base . $k;
+			if ( isset( $plugin_info->response[ $plugin_file ] ) ) {
+				$plugins[ $key ]['slug'] = $plugin_info->response[ $plugin_file ]->slug;
+			} elseif ( isset( $plugin_info->no_update[ $plugin_file ] ) ) {
+				$plugins[ $key ]['slug'] = $plugin_info->no_update[ $plugin_file ]->slug;
 			} else {
-				$plugin_path = $plugin_base . $plugin_file;
-			}
-			$id[] = array( $plugin_file );
-
-			if ( is_object( $update_plugins ) && is_array( $update_plugins->response ) ) {
-				// Check for plugin updates.
-				foreach ( $update_plugins->response as $uk => $uv ) {
-					// If we have a match.
-					if ( $uk === $k ) {
-						$this->debug_log( 'An update exists for: ' . $v['Name'] );
-						// Save the update version.
-						//update_post_meta( $id, 'update', $uv->new_version );
-						// Save the current version.
-						//update_post_meta( $id, 'version', $v['Version'] );
-					}
-				}
+				$plugins[ $key ]['slug'] = false;
 			}
-		}//end foreach
-		return $id;
+		}
+
+		// Compact output.
+		$plugins = array_map(
+			function( $plugin ) {
+				return array(
+					'slug'    => $plugin['slug'],
+					'name'    => $plugin['Name'],
+					'version' => $plugin['Version'],
+					'active'  => $plugin['active'],
+				);
+			},
+			$plugins
+		);
+
+		return apply_filters( 'phpcompat_plugins_to_scan', $plugins );
+	}
+
+	/**
+	 * Get all themes that should be scanned.
+	 *
+	 * @return array
+	 */
+	public function get_themes_to_scan() {
+		$themes_data = wp_prepare_themes_for_js();
+
+		$themes = array_map(
+			function( $theme ) {
+				return array(
+					'slug'    => $theme['id'],
+					'name'    => $theme['name'],
+					'version' => $theme['version'],
+					'active'  => true === $theme['active'] ? 'yes' : 'no',
+				);
+			},
+			$themes_data
+		);
+
+		return apply_filters( 'phpcompat_themes_to_scan', $themes );
```

Comment:
We should ensure we have a proper [docblock](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters) here

Reply:
Docblock added in https://github.com/10up/phpcompat/pull/2/commits/7755905f9a91fe3d760873565e9e0d0abd325e39